core/src/loader.rs

```diff
@@ -1584,7 +1585,92 @@ impl<'gc> Loader<'gc> {
                             &mut activation.context,
                             "complete",
                         );
-                        Avm2::dispatch_event(uc, complete_evt, sound_object);
+                        Avm2::dispatch_event(&mut activation.context, complete_evt, sound_object);
```

Comment:
Have you tested that the `id3` event is always dispatched after the `complete` event, and that `Sound.id3` is still null when accessing it the `complete` event handler?

Reply:
As for mp3, it will be fired before complete event is fired. Actually, id3 event will be fired as soon as the id3 metadata was read completely.

Reply:
Tested with mp3:
progress event
id3 event [object ID3Info]
progress event
complete event